staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go

```diff
@@ -1376,54 +1384,48 @@ func TestNeedsUpdate(t *testing.T) {
 	}, {
 		testName: "If appProtocol is the same",
 		updateFn: func() (oldSvc *v1.Service, newSvc *v1.Service) {
-			oldSvc = newService("tcp-service", v1.ServiceTypeLoadBalancer, tweakAddPorts(v1.ProtocolTCP, 22))
-			newSvc = oldSvc.DeepCopy()
+			oldSvc = newService("tcp-service", v1.ServiceTypeLoadBalancer)
+			newSvc = copyService(oldSvc)
 			return
 		},
 		expectedNeedsUpdate: false,
 	}, {
 		testName: "If service IPFamilies from single stack to dual stack",
 		updateFn: func() (oldSvc *v1.Service, newSvc *v1.Service) {
-			oldSvc = newService("tcp-service", v1.ServiceTypeLoadBalancer, tweakAddPorts(v1.ProtocolTCP, 22), tweakAddAppProtocol("http"), tweakAddIPFamilies(v1.IPv4Protocol))
-			newSvc = oldSvc.DeepCopy()
-			newSvc.Spec.IPFamilies = []v1.IPFamily{v1.IPv4Protocol, v1.IPv6Protocol}
+			oldSvc = newService("tcp-service", v1.ServiceTypeLoadBalancer, tweakAddIPFamilies(v1.IPv4Protocol))
```

Comment:
The tweak should be named "set" not "add"

Reply:
done